internal/evaluation_target/application_snapshot_image/application_snapshot_image.go

```diff
@@ -160,6 +174,64 @@ func (a *ApplicationSnapshotImage) ValidateAttestationSignature(ctx context.Cont
 	return nil
 }
 
+// ValidateAttestationSyntax validates the attestations against known JSON
+// schemas, errors out if there are no attestations to check to prevent
+// sucessful syntax check of no inputs, must invoke
+// [ValidateAttestationSignature] to prefill the attestations.
+func (a ApplicationSnapshotImage) ValidateAttestationSyntax(ctx context.Context) error {
+	if len(a.attestations) == 0 {
+		log.Debug("No attestation data found, possibly due to attestation image signature not being validated beforehand")
+		return EV001
+	}
+
+	allErrors := map[string][]jsonschema.KeyError{}
+	for _, att := range a.attestations {
+		statementBytes, _, err := statementFrom(ctx, att)
+		if err != nil {
+			return EV002.CausedBy(err)
+		}
+		if statementBytes == nil {
+			return EV002.CausedByF("Unrecognized statement JSON")
+		}
+
+		// at least one of the schemas needs to pass validation
+		for id, schema := range attestationSchemas {
+			if errs, err := schema.ValidateBytes(ctx, statementBytes); err != nil {
```

Comment:
Not sure if I'm doing something wrong, but I have an image that has attestations with invalid buildTypes. Should I expect a schema validation error? It looks like it gets all the way to conftest before an error is reached.

```
🐚 go run main.go validate image --policy "$(cat policy.json)" --image quay.io/lucarval/minimal-container/min:latest2
go: downloading github.com/qri-io/jsonschema v0.2.1
go: downloading github.com/qri-io/jsonpointer v0.1.1
Error: 1 error occurred:
	* error validating image quay.io/lucarval/minimal-container/min:latest2 of component Unnamed: parse files: no files found


exit status 1

🐚 cat policy.json 
{
  "description": "ACME & co policy",
  "sources": [
    "git::https://github.com/hacbs-contract/ec-policies//policy"
  ],
  "publicKey": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEa59xcJ1ZuApObPRqy4fAZP43l5up\nVMJLBAROjLUlXlCwJ7iAPJk9syUrx1FFGR6nRqxUeqXXXRbbjBoquqrzkg==\n-----END PUBLIC KEY-----\n" ,
  "configuration": {
    "excludeRules": [
    ]
  }
}

```

Reply:
Forgot to update output for syntax issues, should be fixed now